### PR TITLE
Add sideset assembly support.

### LINKED ISSUE #16: .side_set N opt is not currently implemented
The optional `opt` flag for .side_set is not currently implemented in the assembler. This is used in some of the common PIO examples and an example in section 3.5.4 of the RP2040 data sheet.

### LINKED ISSUE #21: Example pwm.pio re-implementation not working
I've had my try at replicating the [pwm.pio example](https://github.com/raspberrypi/pico-examples/tree/master/pio/pwm) from the raspberry pi GitHub.

The C-Code ported to CircuitPython does not work, and generates different pio assembly code than if rewritten in MicroPyton.

[Pwm_Example_MP_CP.zip](https://github.com/adafruit/Adafruit_CircuitPython_PIOASM/files/6266111/Pwm_Example_MP_CP.zip)

One notable difference is in the generated pio binary code:
```
CP: [32896, 40999, 41030, 165, 4102, 41026, 131]
MP: [36992, 40999, 41030, 165, 6150, 41026, 131]
```

Looking at a disassembled version of these numbers, it can be seen, that the sideset arguments are different between both versions.
```
CircuitPython
PUSH    noblock 00000 <----------
MOV     X OSR   00000
MOV     Y ISR   00000
JMP     X!=Y 5  00000
JMP      6      10000 <----------
MOV     Y Y     00000
JMP     Y-- 3   00000

Micropython
PUSH    noblock 10000 <----------
MOV     X OSR   00000
MOV     Y ISR   00000
JMP     X!=Y 5  00000
JMP      6      11000 <----------
MOV     Y Y     00000
JMP     Y-- 3   00000
```

That's just one example, I've also had troubles getting other pio code to run, but this was the most understandable for debugging.

### PR DESCRIPTION
**Pull request by tannewt**
You must also set `sideset_enable` when creating the rp2pio.StateMachine
object.

Fixes #16 and fixes #21.

This needs https://github.com/adafruit/circuitpython/pull/5766

### CODE REVIEW

**Comment by tannewt** [comment]
@dannystaple Please try this out. Note that it'll need a new CircuitPython build as well.

**Comment by dannystaple** [comment]
Which build? Will the 7.1.0-beta-3 build do this or do I need to pick up a CI artefact for it?

**Comment by dannystaple** [comment]
Ok - not sure this is working. Perhaps I'm missing something.
My code:
```
.program pwm_prog
.wrap_target
.side_set 1 opt                ; 1 sideset pin, optional.

    set x 5           side 0
    set y 10         
pwmloop:
    jmp x!=y skip           ; skip if they are different
    nop             side 1  ; drive side set pin high (which will be the one time they are the same)

skip:
    jmp y-- pwmloop         ; decrement x and go to pwm loop.
.wrap
```

(I know we can ignore wrap/wrap target)

The output assembled:
```
000 111_10000_001_00101
001 111_00000_010_01010
010 000_00000_101_00100
011 101_11000_010_00010
100 000_00000_100_00010
```

I think I am expecting:
```
000 111_10001_001_00101 
001 111_00001_010_01010 
010 000_00001_101_00100 
011 101_11000_010_00010 
100 000_00001_100_00010 
```

I could be wrong though.

I have used the 7.1.0-beta3, and just to be sure I was loading your code, called the module adafruit_pioasm_tst, and imported that. Assembly:

```pwm_prog = adafruit_pioasm_tst.assemble(pwm_prog_raw)```

FYI - the output I made with this, on the Pico:
```
instructions = ('{:016b}'.format(inst) for inst in pwm_prog)
formatted = ("{i}_{ss}_{src}_{dst}".format(
    i=inst[0:3],
    ss=inst[3:8],
    src=inst[8:11],
    dst=inst[11:16]
) for inst in instructions)

numbered =  ['{:03b} {}'.format(line, inst) for line, inst in enumerate(formatted)]

print('\n'.join(numbered))
```

**Comment by dannystaple** [comment]
Wait - being a donut. I made those expected things based on your unit tests. The last bit shoudln't be set - that was because you had a [1] in the test code I used to get example bits from. I might be getting what I mean.

**Comment by dannystaple** [comment]
Confirmed - output with the old library:
```
000 111_00000_001_00101
001 111_00000_010_01010
010 000_00000_101_00100
011 101_10000_010_00010
100 000_00000_100_00010
```
It's subtle - but it's the 1 for side set 0 and 11 for side set 1 that makes the difference. Time to get the scope out and check that it does what I think it should (it's a really stupid fixed pwm).

File: tests/testpioasm.py [hunk 1]
```diff
@@ -44,6 +44,13 @@ def testNop(self):
         self.assertAssemblesTo(".side_set 1\nnop side 1", [0b101_10000_010_00_010])
         self.assertAssemblesTo(".side_set 1\nnop side 1 [1]", [0b101_10001_010_00_010])
 
+    def testSidesetOpt(self):
+        self.assertAssemblesTo(".side_set 1 opt\nnop side 1", [0b101_11000_010_00_010])
```
Comment (dannystaple) [inline tests/testpioasm.py:48]: Is it worth having a test for side 0 without the timing option too? Presumably with the 0b101_10000_010_00_010

Comment (tannewt) [reply tests/testpioasm.py:48]: Can't hurt. Added!

**Comment by tannewt** [comment]
Absolute newest builds should work with this now.

**Review by dannystaple — APPROVED** [review]
Looks good to me. It tests out with the right bytecode, comes with tests and example code.

**Review by jepler — APPROVED** [review]
Thank you!

**Pull request merged**